tor_flair/main.py

```diff
@@ -0,0 +1,114 @@
+from tor_core.initialize import build_bot
+
+from tor_flair import __version__
+
+settings_location = 'tor-flair/config/settings'
+instructions_location = 'tor-flair/config/instructions'
+
+instructions_page = (
+    'Thanks for using the Transcribers of Reddit flair helper bot! Our goal '
+    'is to make your day a little easier by automating the process of identifying '
+    'our volunteers on your subreddit.\n\n'
+    'Simply set flair-text and / or flair-css in {} '
+    'and u/tor_flair_bot will get the changes the next time a volunteer accepts a '
+    'job for your subreddit.\n\n'
+    'If you have any questions or issues with the bot, please '
+    'send a modmail to r/TranscribersOfReddit and we\'ll get you sorted as quickly '
+    'as we can!'.format(settings_location)
+)
+
+
+settings_page = (
+    '; Set the flair text and css class below to whatever you want.  \n'
+    '; Both options are optional - whichever one is blank be left alone.  \n'
+    '; Example:  \n'
+    ';  \n'
+    '; flair-text: darn cool person  \n'
+    '; flair-css: coolperson  \n'
+    '\n\n'
+    'flair-text:\n\n'
+    'flair-css:'
+)
+
+
+def setup_wiki(subreddit):
+    subreddit.wiki.create(settings_location, content=settings_page)
+    subreddit.wiki.create(instructions_location, content=instructions_page)
+
+
+def get_config(subreddit):
+    return get_wiki_page(settings_location, config, subreddit=subreddit)
+
+
+def check_inbox(config):
+    logging.debug('Checking inbox...')
+    for item in config.r.inbox.unread():
+        if 'invitation to moderate' in item.subject.lower():
+            logging.info(
+                'Received offer to moderate {}!'.format(item.subreddit.display_name)
+            )
+            item.subreddit.mod.accept_invite()
+            setup_wiki(item.subreddit)
+            config.redis.sadd('flair_subreddits', item.subreddit.display_name)
+            item.mark_read()
+        if 'has been removed as a moderator from' in item.subject.lower():
+            logging.warning(
+                'I\'ve been kicked from {}!'.format(item.subreddit.display_name)
+            )
+            config.redis.srem('flair_subreddits', item.subreddit.display_name)
+            item.mark_read()
+
+
+def run(config):
+    check_inbox(config)
+    new_job = config.redis.lpop('flair_queue')
+    if not new_job:
+        time.sleep(10)
+        return
+    # format: user::subreddit
+    new_job = new_job.decode()  # bytes to str
+    user = config.r.redditor(
+        new_job[:new_job.index('::')]
+    )
+    subreddit = config.r.subreddit(new_job[new_job.index('::')+2:])
+
+    logging.info(
+        'Setting flair for {} in {}...'.format(user.name, subreddit.display_name)
+    )
+
+    subreddit_config = get_config(subreddit).split('\r\n')
+    subreddit_config = [y for y in subreddit_config if not y.startswith(';') and y != '']
+
+    text = subreddit_config[0][11:].strip()
+    css = subreddit_config[1][10:].strip()
```

Comment:
This works but is not idiot-proof. If subreddits don't follow the exact format provided in the default config, it will yield unintended results or even break the bot if there are no non-empty non-comment lines. Even though it's a bit more work I would suggest creating a separate method that reads a config wiki page and turns it into a dict, which would also be helpful later on if we add post flairs or other configurable stuff.

Reply:
Yeah, that's a good idea. :+1: